### PR TITLE
Activity-based prebuilds

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Starting with this PR, when you enable prebuilds on a repository, there no longer will be a webhook installed on the underlying scm-hosted repo. Instead, prebuilds will be triggered based on workspace activity, or in other words, a prebuild will trigger any time a workspace is created on a repo which has them enabled. This of course still takes prebuild preferences into account, so commit intervals, branch matching strategies and other settings still apply.

This is how the setup will look like for current repos:

```mermaid
graph TD
    subgraph GitHub
        GHRepo[Repository]
    end
    subgraph Gitpod
        GPRepo[Configured Repository]
        Workspace[Workspace]
    end

    GHRepo -- Webhook --> GPRepo
    GHRepo -- Code --> Workspace
    GPRepo -- Code push Trigger --> Prebuild[Prebuild]
    Workspace -- Workspace Creation Trigger --> Prebuild
```

And for new repositories, without purely activity-based

```mermaid
graph TD
    subgraph GitHub
        GHRepo[Repository]
    end
    subgraph Gitpod
        GPRepo[Configured Repository]
        Workspace[Workspace]
    end

    GHRepo -- Code --> Workspace
    Workspace -- Workspace Creation Trigger --> Prebuild
```

## Related Issue(s)

Fixes ENT-435

## How to test

In order to test, you can follow these steps:

1. In the [preview environment](https://ft-activit0b1202b619.preview.gitpod-dev.com/workspaces), configure a repository under `Repository settings`. It should preferably be some throwaway repo to which you can push to (for instance, feel free to fork https://github.com/filiptronicek/test-foofies/ from which you can run `./commits.sh <n>`)
2. Enable prebuilds for the repository and create a new workspace with it
3. You should see a new prebuild being triggered after your workspace has started
4. Try creating another workspace from the same repo to see that:
	a. The prebuild was used for the workspace
	b. No new prebuild was triggered
5. Now, push 21 commits to your repo and start a new workspace on it, it should trigger a prebuild

## Documentation

https://github.com/gitpod-io/website/pull/4756

#### Preview status

https://ft-activit0b1202b619.preview.gitpod-dev.com/workspaces

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/repositories/detail/prebuilds/PrebuildSettingsForm.tsx [hunk 1]
```diff
@@ -17,15 +21,26 @@ import { LoadingButton } from "@podkit/buttons/LoadingButton";
...
+import { trackEvent } from "../../../Analytics";
+import dayjs from "dayjs";
 
 const DEFAULT_PREBUILD_COMMIT_INTERVAL = 20;
 
 type Props = {
     configuration: Configuration;
 };
 
+const COACHMARK_KEY = "new_prebuilds_trigger_notification";
```
Comment (filiptronicek) [inline components/dashboard/src/repositories/detail/prebuilds/PrebuildSettingsForm.tsx:36]: It's not really a coach mark, but I for some reason couldn't imagine `coachmarksDismissals` being used for anything else 🤦

Comment (filiptronicek) [reply components/dashboard/src/repositories/detail/prebuilds/PrebuildSettingsForm.tsx:36]: Well, hindsight's 20/20

File: components/public-api/gitpod/v1/configuration.proto [hunk 2]
```diff
@@ -19,12 +19,20 @@ message Configuration {
   WorkspaceSettings workspace_settings = 7;
 }
 
+enum PrebuildActivationStrategy {
```
Comment (svenefftinge) [inline components/public-api/gitpod/v1/configuration.proto:22]: nit: It's not the startegy for "activating" prebuilds, but for triggering prebuilds.

**Review by svenefftinge — APPROVED** [review]
Sweet! ❤️  Love how much complexity this removes

**Comment by filiptronicek** [comment]
:shipit:

**Pull request merged**